utils/coins.go

```diff
@@ -17,14 +17,15 @@ func CoinFromRequestKey(query []byte, accAddr sdk.AccAddress) (sdk.Coin, error)
 }
 
 func DenomFromRequestKey(query []byte, accAddr sdk.AccAddress) (string, error) {
-	idx := bytes.Index(query, accAddr)
-	if idx == -1 {
-		return "", errors.New("AccountBalanceCallback: invalid request query")
-	}
-	denom := string(query[idx+len(accAddr):])
-	if err := sdk.ValidateDenom(denom); err != nil {
+	balancesStore := query[1:]
+	accAddr2, denom, err := banktypes.AddressAndDenomFromBalancesStore(balancesStore)
```

Comment:
Let's add a test for this function.  Should cover:
- valid case,
- error in `AddressAndDenomFromBalancesStore`
- not matching account addressses

Reply:
added